### PR TITLE
implement-creating-and-exporting-users-using-files

### PR DESCRIPTION
**Pull request by Stan3slaw**
implemented creating users from file and exporting users to file

### CODE REVIEW

File: src/users/users.service.js [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+    mockedUsers.push(...normalizedUsers);
+  } else {
+    normalizedUsers = { ...users, id: Number(users.id) };
+    mockedUsers.push(normalizedUsers);
+  }
+
+  return normalizedUsers;
+};
+
+const exportUsersToXml = (_, res, usersInXml, filePath) => {
```
Comment (bohdan-shtohrinets-sigma) [inline src/users/users.service.js]: @Stan3slaw why you ignore firts arg? If it is not needed why you include it here?

File: src/users/users.service.js [hunk 2]
```diff
@@ -0,0 +1,66 @@
+const fs = require('fs');
+const path = require('path');
+
```
Comment (bohdan-shtohrinets-sigma) [inline src/users/users.service.js]: @Stan3slaw better to use classes

File: src/users/users.controller.js [hunk 3]
```diff
@@ -1,20 +1,41 @@
-const users = require('../core/mock-data');
+const xml2js = require('xml2js');
+
+const usersService = require('./users.service');
+const mockedUsers = require('../core/mock-data');
 
 const find = (req, res) => {
```
Comment (bohdan-shtohrinets-sigma) [inline src/users/users.controller.js]: @Stan3slaw better to use classes

File: src/users/users.service.js [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+      rs.on('data', (chunk) => {
+        data += chunk;
+      });
+
+      rs.on('error', (err) => {
+        console.error('read stream error', err);
+      });
+
+      rs.on('end', () => {
+        res.writeHead(200, {
```
Comment (bohdan-shtohrinets-sigma) [inline src/users/users.service.js]: better to do this logic inside controller, services are usually used for business logic management and controllers for resp forming

**Review by bohdan-shtohrinets-sigma — CHANGES_REQUESTED** [review]

File: src/users/users.service.js (hunk 1 above)
Comment (Stan3slaw) [reply src/users/users.service.js]: Okay, I will remove it

File: src/users/users.service.js (hunk 4 above)
Comment (Stan3slaw) [reply src/users/users.service.js]: Yeah, I know this stuff, but in this case logic of exporting users to file is strongly bound, so I can't just move this peace of code to the controller.

File: src/users/users.controller.js (hunk 3 above)
Comment (Stan3slaw) [reply src/users/users.controller.js]: Okay, I will fix it.

File: src/users/users.service.js (hunk 2 above)
Comment (Stan3slaw) [reply src/users/users.service.js]: Okay, I will fix it.

File: src/users/users.service.js (hunk 4 above)
Comment (bohdan-shtohrinets-sigma) [reply src/users/users.service.js]: @Stan3slaw you can try to save your data to some variable and return it in success case. Otherwise throw error.

**Review by bohdan-shtohrinets-sigma — APPROVED** [review]

**Pull request merged**